crates/kitsune_p2p/dht/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+//! Defines the structure of the DHT, the objects which inhabit it, and the operations
+//! on these objects.
+//!
+//! ## Structure and contents of the DHT
+//!
+//! The DHT is populated by Agents and Ops. Agents introduce new Ops to the DHT,
+//! and are responsible for storing their own Ops and the Ops of other Agents.
+//!
+//! Each Agent has a fixed Location, and each Op has both a fixed
+//! Location and a fixed Timestamp.
+//!
+//! Locations are one-dimensional, and are best thought of as positions
+//! on a circle. They are represented by `Wrapping<u32>`.
+//! Timestamps are best thought of as positions on a timeline, and are represented
+//! by an i64 which denotes microseconds since the UNIX epoch.
+//! The Location dimension is circular, and the Time dimension is linear.
+//!
+//! Each Agent has an Arc which marks out a portion of the circle of Locations.
```

Comment:
Well, the correct geometrical term is indeed "arc". It would feel like a circular definition (no pun intended) to describe an Arc in terms of arcs. But maybe this should be explicitly mentioned :) A [segment](https://en.wikipedia.org/wiki/Circular_segment) is a different geometrical concept, so I wanted to use a geometrically neutral word here.